### PR TITLE
Feat/more e2e tests

### PR DESCRIPTION
**Pull request by ilyasymbiotic**
### **PR Type**
Tests

___

### **Description**
- Add proof sync e2e test

- Select signer-only node dynamically

- Stop/start container to test syncing

- Validate proof availability post-restart

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Load deployment data"] -- "create EVM client" --> B["Get current & next epoch"]
  B -- "derive next valset" --> C["Find signer-only node"]
  C -- "stop container" --> D["Wait for next epoch"]
  D -- "other node proves" --> E["Fetch metadata (requestId)"]
  E -- "restart signer-only" --> F["Wait healthy"]
  F -- "GetAggregationProof(requestId)" --> G["Proof synced on restarted signer"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>sync_test.go</strong><dd><code>Add proof sync e2e test with signer restart</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

e2e/tests/sync_test.go

<ul><li>Import <code>github.com/samber/lo</code> helper.<br> <li> Add <code>TestAggregatorProofSync</code> end-to-end test.<br> <li> Dynamically pick signer-only node, stop/start container.<br> <li> Verify proof sync via gRPC after restart.</ul>

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/314/files#diff-4a40b2f4481adacb8454df1cb5b7776bd45b78472b484a8d9fcf718a1f24afdd">+92/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: e2e/tests/settlement_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		t.Logf("Reached epoch %d", oneSettlementEpoch)
+
+		nextCaptureTimestamp, err := evmClient.GetEpochStart(t.Context(), oneSettlementEpoch)
+		require.NoError(t, err, "Failed to get epoch start timestamp")
+		t.Logf("Epoch %d start timestamp: %d", oneSettlementEpoch, nextCaptureTimestamp)
+
+		oneSettlementConfig, err = evmClient.GetConfig(t.Context(), nextCaptureTimestamp, oneSettlementEpoch)
+		require.NoError(t, err, "Failed to get network config")
+
+		require.Len(t, oneSettlementConfig.Settlements, 1, "Expected exactly one settlement after removal")
```
Comment (oxsteins) [inline e2e/tests/settlement_test.go]: We should leave scope for some race condition where we remove settlement but it's executed after the next epoch. let's give it 2 epochs for ensuring consistency even in edge case scenarios.

File: e2e/tests/sync_test.go [hunk 2]
```diff
@@ -189,14 +283,22 @@ func waitForHealthy(ctx context.Context, healthURL string, timeout time.Duration
 // createEVMClient creates an EVM client for interacting with the blockchain
 func createEVMClient(t *testing.T, deploymentData RelayContractsData) *evm.Client {
 	t.Helper()
+	kp, err := keyprovider.NewSimpleKeystoreProvider()
+	require.NoError(t, err)
+	// private key is from /e2e/contracts/network-scripts/deploy.sh
+	privateKey, err := crypto.NewPrivateKey(symbiotic.KeyTypeEcdsaSecp256k1, common.Hex2Bytes("ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"))
```
Comment (oxsteins) [inline e2e/tests/sync_test.go]: let's set this as a const

File: e2e/tests/settlement_test.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+	t.Run("set genesis with re-added settlement", func(t *testing.T) {
+		t.Log("Step 3: Setting genesis for re-added settlement chain")
+
+		client := getGRPCClient(t, 0)
+		t.Log("Waiting for validator set metadata to be available")
+		var nextMetadata *apiv1.GetValidatorSetMetadataResponse
+		err = waitForErrorIsNil(t.Context(), time.Second*30, func() error {
+			nextMetadata, err = client.GetValidatorSetMetadata(t.Context(), &apiv1.GetValidatorSetMetadataRequest{
+				Epoch: (*uint64)(&oneSettlementEpoch),
```
[the remark this answers is not in the document]
Comment (ilyasymbiotic) [reply e2e/tests/settlement_test.go]: fixed

**Review by oxsteins — APPROVED** [review]
